### ISSUE TITLE
[BUG] v1.86.2 broken - failed to watch/ list resources

### ISSUE THREAD

**Issue by cberge908** [kiali/kiali#7506]
### Describe the bug

Since the update to Kiali v1.86.2 the pod seems to crash constantly.
I seems that all the CRDS could not be listed/ watched by Kiali. This was not the case with v1.86.0.
When switching back to v1.86.0, the pod is coming up normally. 

Here is an expect from the pod log on v1.86.2, this is repeated for most other resources:

```shell
stream logs failed container "kiali" in pod "kiali-95b9cd4cf-n55d9" is waiting to start: PodInitializing for istio-system/kiali-95b9cd4cf-n55d9 (kiali)
kiali 2024/07/03 09:47:21 maxprocs: Leaving GOMAXPROCS=8: CPU quota undefined
kiali 2024-07-03T09:47:21Z INF Kiali: Version: v1.86.2, Commit: e44498e9cab23ade6b82b1033d9860d1c32ecc19, Go: 1.22.1
kiali 2024-07-03T09:47:22Z INF Using authentication strategy [openid]
kiali 2024-07-03T09:47:22Z INF Some validation errors will be ignored [KIA1301]. If these errors do occur, they will still be logged. If you think the validation errors you see are incorrect, please report them to the Kiali team if you have not done so already and provide the details of your scenario. This will keep Kiali validations strong for the whole community.
kiali 2024-07-03T09:47:22Z INF Initializing Kiali Cache
kiali 2024-07-03T09:47:22Z INF [Kiali Cache] Waiting for cluster-scoped cache to sync
kiali W0703 09:47:22.233193       1 reflector.go:547] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *v1.ServiceEntry: the server could not find the requested resource (get serviceentries.networking.istio.io)
kiali E0703 09:47:22.233252       1 reflector.go:150] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *v1.ServiceEntry: failed to list *v1.ServiceEntry: the server could not find the requested resource (get serviceentries.networking.istio.io)
kiali W0703 09:47:22.233990       1 reflector.go:547] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *v1.Sidecar: the server could not find the requested resource (get sidecars.networking.istio.io)
kiali E0703 09:47:22.234045       1 reflector.go:150] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *v1.Sidecar: failed to list *v1.Sidecar: the server could not find the requested resource (get sidecars.networking.istio.io)
kiali W0703 09:47:22.257172       1 reflector.go:547] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *v1.PeerAuthentication: the server could not find the requested resource (get peerauthentications.security.istio.io)
kiali W0703 09:47:22.257705       1 reflector.go:547] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *v1.DestinationRule: the server could not find the requested resource (get destinationrules.networking.istio.io)
kiali W0703 09:47:22.257711       1 reflector.go:547] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *v1.WorkloadEntry: the server could not find the requested resource (get workloadentries.networking.istio.io)
kiali E0703 09:47:22.257735       1 reflector.go:150] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *v1.DestinationRule: failed to list *v1.DestinationRule: the server could not find the requested resource (get destinationrules.networking.istio.io)
kiali E0703 09:47:22.257738       1 reflector.go:150] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *v1.WorkloadEntry: failed to list *v1.WorkloadEntry: the server could not find the requested resource (get workloadentries.networking.istio.io)
kiali E0703 09:47:22.257711       1 reflector.go:150] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *v1.PeerAuthentication: failed to list *v1.PeerAuthentication: the server could not find the requested resource (get peerauthentications.security.istio.io)
kiali W0703 09:47:22.257189       1 reflector.go:547] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *v1.WorkloadGroup: the server could not find the requested resource (get workloadgroups.networking.istio.io)
kiali E0703 09:47:22.257759       1 reflector.go:150] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *v1.WorkloadGroup: failed to list *v1.WorkloadGroup: the server could not find the requested resource (get workloadgroups.networking.istio.io)
kiali W0703 09:47:22.257800       1 reflector.go:547] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *v1.VirtualService: the server could not find the requested resource (get virtualservices.networking.istio.io)
kiali E0703 09:47:22.257815       1 reflector.go:150] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *v1.VirtualService: failed to list *v1.VirtualService: the server could not find the requested resource (get virtualservices.networking.istio.io)
kiali W0703 09:47:22.257962       1 reflector.go:547] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *v1.Telemetry: the server could not find the requested resource (get telemetries.telemetry.istio.io)
kiali E0703 09:47:22.257980       1 reflector.go:150] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *v1.Telemetry: failed to list *v1.Telemetry: the server could not find the requested resource (get telemetries.telemetry.istio.io)
kiali W0703 09:47:22.258004       1 reflector.go:547] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *v1.Gateway: the server could not find the requested resource (get gateways.networking.istio.io)
kiali E0703 09:47:22.258021       1 reflector.go:150] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *v1.Gateway: failed to list *v1.Gateway: the server could not find the requested resource (get gateways.networking.istio.io)
kiali W0703 09:47:23.101530       1 reflector.go:547] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *v1.VirtualService: the server could not find the requested resource (get virtualservices.networking.istio.io)
kiali E0703 09:47:23.101557       1 reflector.go:150] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *v1.VirtualService: failed to list *v1.VirtualService: the server could not find the requested resource (get virtualservices.networking.istio.io)
kiali W0703 09:47:23.102237       1 reflector.go:547] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *v1.Sidecar: the server could not find the requested resource (get sidecars.networking.istio.io)
kiali E0703 09:47:23.102252       1 reflector.go:150] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *v1.Sidecar: failed to list *v1.Sidecar: the server could not find the requested resource (get sidecars.networking.istio.io)
kiali W0703 09:47:23.283838       1 reflector.go:547] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *v1.Gateway: the server could not find the requested resource (get gateways.networking.istio.io)
kiali E0703 09:47:23.283866       1 reflector.go:150] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *v1.Gateway: failed to list *v1.Gateway: the server could not find the requested resource (get gateways.networking.istio.io)
kiali W0703 09:47:23.328547       1 reflector.go:547] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *v1.WorkloadEntry: the server could not find the requested resource (get workloadentries.networking.istio.io)
kiali E0703 09:47:23.328575       1 reflector.go:150] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *v1.WorkloadEntry: failed to list *v1.WorkloadEntry: the server could not find the requested resource (get workloadentries.networking.istio.io)
kiali W0703 09:47:23.410650       1 reflector.go:547] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *v1.ServiceEntry: the server could not find the requested resource (get serviceentries.networking.istio.io)
kiali E0703 09:47:23.410720       1 reflector.go:150] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *v1.ServiceEntry: failed to list *v1.ServiceEntry: the server could not find the requested resource (get serviceentries.networking.istio.io)
kiali W0703 09:47:23.419704       1 reflector.go:547] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *v1.DestinationRule: the server could not find the requested resource (get destinationrules.networking.istio.io)
kiali E0703 09:47:23.419754       1 reflector.go:150] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *v1.DestinationRule: failed to list *v1.DestinationRule: the server could not find the requested resource (get destinationrules.networking.istio.io)
kiali W0703 09:47:23.627128       1 reflector.go:547] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *v1.Telemetry: the server could not find the requested resource (get telemetries.telemetry.istio.io)
kiali E0703 09:47:23.627172       1 reflector.go:150] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *v1.Telemetry: failed to list *v1.Telemetry: the server could not find the requested resource (get telemetries.telemetry.istio.io)
kiali W0703 09:47:23.684179       1 reflector.go:547] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *v1.PeerAuthentication: the server could not find the requested resource (get peerauthentications.security.istio.io)
kiali E0703 09:47:23.684204       1 reflector.go:150] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *v1.PeerAuthentication: failed to list *v1.PeerAuthentication: the server could not find the requested resource (get peerauthentications.security.istio.io)
kiali W0703 09:47:23.693786       1 reflector.go:547] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *v1.WorkloadGroup: the server could not find the requested resource (get workloadgroups.networking.istio.io)
kiali E0703 09:47:23.693803       1 reflector.go:150] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *v1.WorkloadGroup: failed to list *v1.WorkloadGroup: the server could not find the requested resource (get workloadgroups.networking.istio.io)
kiali W0703 09:47:25.234494       1 reflector.go:547] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *v1.VirtualService: the server could not find the requested resource (get virtualservices.networking.istio.io)
kiali E0703 09:47:25.234522       1 reflector.go:150] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *v1.VirtualService: failed to list *v1.VirtualService: the server could not find the requested resource (get virtualservices.networking.istio.io)
kiali W0703 09:47:25.257420       1 reflector.go:547] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *v1.Gateway: the server could not find the requested resource (get gateways.networking.istio.io)
kiali E0703 09:47:25.257449       1 reflector.go:150] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *v1.Gateway: failed to list *v1.Gateway: the server could not find the requested resource (get gateways.networking.istio.io)
kiali W0703 09:47:25.258177       1 reflector.go:547] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *v1.Sidecar: the server could not find the requested resource (get sidecars.networking.istio.io)
kiali E0703 09:47:25.258204       1 reflector.go:150] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *v1.Sidecar: failed to list *v1.Sidecar: the server could not find the requested resource (get sidecars.networking.istio.io)
kiali W0703 09:47:25.718706       1 reflector.go:547] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *v1.WorkloadEntry: the server could not find the requested resource (get workloadentries.networking.istio.io)
kiali E0703 09:47:25.718735       1 reflector.go:150] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *v1.WorkloadEntry: failed to list *v1.WorkloadEntry: the server could not find the requested resource (get workloadentries.networking.istio.io)
kiali W0703 09:47:25.866416       1 reflector.go:547] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *v1.DestinationRule: the server could not find the requested resource (get destinationrules.networking.istio.io)
kiali E0703 09:47:25.866450       1 reflector.go:150] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *v1.DestinationRule: failed to list *v1.DestinationRule: the server could not find the requested resource (get destinationrules.networking.istio.io)
kiali W0703 09:47:26.035315       1 reflector.go:547] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *v1.ServiceEntry: the server could not find the requested resource (get serviceentries.networking.istio.io)
kiali E0703 09:47:26.035351       1 reflector.go:150] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *v1.ServiceEntry: failed to list *v1.ServiceEntry: the server could not find the requested resource (get serviceentries.networking.istio.io)
kiali W0703 09:47:26.052349       1 reflector.go:547] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *v1.PeerAuthentication: the server could not find the requested resource (get peerauthentications.security.istio.io)
kiali E0703 09:47:26.052389       1 reflector.go:150] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *v1.PeerAuthentication: failed to list *v1.PeerAuthentication: the server could not find the requested resource (get peerauthentications.security.istio.io)
kiali W0703 09:47:26.071056       1 reflector.go:547] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *v1.WorkloadGroup: the server could not find the requested resource (get workloadgroups.networking.istio.io)
kiali E0703 09:47:26.071075       1 reflector.go:150] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *v1.WorkloadGroup: failed to list *v1.WorkloadGroup: the server could not find the requested resource (get workloadgroups.networking.istio.io)
kiali W0703 09:47:26.310321       1 reflector.go:547] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *v1.Telemetry: the server could not find the requested resource (get telemetries.telemetry.istio.io)
kiali E0703 09:47:26.310350       1 reflector.go:150] pkg/mod/k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *v1.Telemetry: failed to list *v1.Telemetry: the server could not find the requested resource (get telemetries.telemetry.istio.io)
```

### Expected Behavior

Kiali starts as usual.

### What are the steps to reproduce this bug?

1. Update to v1.86.2
2. Kiali pod fails to start 

### Environment
*Learn about how to determine versions [here](https://kiali.io/docs/faq/general/#how-do-i-determine-what-version-i-am-running).*

- **Kiali version:** v1.86.2
- **Istio version:** v1.22.2
- **Kubernetes impl:** AKS
- **Kubernetes version:** v1.29.4
- **Other notable environmental factors:** We're not using ambient mode (normal sidecars). So from the fixes that were introduced with v1.86.2 we should not be affected.

**Comment by nrfox** [comment]
Do you have the v1 istio CRDs installed in your cluster?

**Comment by cberge908** [comment]
> Do you have the v1 istio CRDs installed in your cluster?

Interesting... I had always kept the `istio/base` chart in sync with the `istio/istiod` chart. I somehow was expecting that the CRDs are updated automatically with the `istio/base` chart. I'm just reading now that updating CRDs via helm in not supported 😐

I've applied the updated CRDs manually, obviously they are now on `v1` and Kiali v1.86.2 works without throwing any additional errors. So my fault... issue can be closed - thanks @nrfox for the hint! 🤝

**Comment by nrfox** [comment]
> I'm just reading now that updating CRDs via helm in not supported

I didn't realize that either

**Comment by cberge908** [comment]
Just for completeness: https://helm.sh/docs/chart_best_practices/custom_resource_definitions/#some-caveats-and-explanations

> There is no support at this time for upgrading or deleting CRDs using Helm. This was an explicit decision after much community discussion due to the danger for unintentional data loss. Furthermore, there is currently no community consensus around how to handle CRDs and their lifecycle. As this evolves, Helm will add support for those use cases.

**Comment by madduci** [comment]
Hi

I've just upgraded the Kiali version from 1.86.0 to 1.86.2 in a local KIND Cluster running with Istio 1.20.8 and i get the same error. AFAIK, the newer CRDs are only available on newer Istio Versions, here are all the CRDs available for 1.20.8 and the `v1` aren't yet there.

I had to downgrade the version of Kiali to get it work correctly again. Any Hint on how could i solve the issue?
